R/pkgdown.R

```diff
@@ -30,25 +29,31 @@ use_pkgdown <- function(config_file = "_pkgdown.yml", destdir = "docs") {
   check_is_package("use_pkgdown()")
   check_installed("pkgdown")
 
-  use_build_ignore(c(config_file, destdir))
-  use_build_ignore("pkgdown")
+  use_build_ignore(c(config_file, destdir, "pkgdown"))
   use_git_ignore(destdir)
 
-  ui_todo("
-    Record your site's {ui_field('url')} in the pkgdown config file \\
-    (optional, but recommended)")
+  config <- pkgdown_config(destdir)
+  config_path <- proj_path(config_file)
+  write_over(config_path, yaml::as.yaml(config))
+  edit_file(config_path)
 
-  if (has_logo()) {
-    pkgdown_build_favicons(proj_get(), overwrite = TRUE)
```

Comment:
Is favicon creation being done by pkgdown itself now?

Because if so, there's some more stuff we can remove when we delete this:

``` r
# usethis itself should not depend on pkgdown
# all usage of this wrapper is guarded by `check_installed("pkgdown")`
pkgdown_build_favicons <- function(...) {
  get("build_favicons", asNamespace("pkgdown"), mode = "function")(...)
}
```

(There is one remaining usage in `use_pkgdown_travis()` btw, which we should also remove.)

And `has_logo()` can probably go. And some mocks in tests ....

Reply:
If you confirm that usethis is out of the favicon creation business, I can do all the other cleanup I refer to above.

Reply:
Yeah, usethis should definitely be out of the favicon creation business as pkgdown now does it (if needed) when you build the site.

Reply:
OK I'm about to push a commit deleting more code.